work/builder/builder.go

```diff
@@ -280,5 +296,22 @@ func ExtractBundlesAndIncorporate(arrayTxs []*types.Transaction, txBundlingModul
 		return flattenedTxs, nil
 	}
 
+	// Set correct target tx hash of bundles
+	// This is necessary since tx from auction module is not included in the arrayTxs but is included in the incorporatedTxs
+	bundles = SetCorrectTargetTxHash(bundles, incorporatedTxs)
```

Comment:
Won't the bid transaction be executed normally without TxOrGen being deleted from incorporatedTxs?

Reply:
Could you please add these cases to the coordinateTargetTxHash test?
- [required true bundle, required false bundle]
- [required false bundle]

Reply:
I added more tests [b04cad2](https://github.com/kaiachain/kaia/pull/466/commits/b04cad2b84bf664fe70f25dffb4f798c389d4dca)

I also found the logical error. Now two bundles can have dependency to one target.
- `bundle.Has(tx)`: Gasless
- `budle.TargetRequired && bundle.Target == tx`: Auction

But current `FindDependentBundleIdx` only returns one bundle so that Auction bundle will be ignored and wrongly inserted in a block. I fixed this and also improve logic in 0101508